src/main/webapp/script.js

```diff
@@ -30,7 +30,20 @@ async function getAuthToken() {
 }
 
 function getIdFromURL(url) {
-  var start = url.lastIndexOf('/') + 1;
-  var end = url.indexOf('?', start);
-  return url.slice(start, end);
+  if (url.search("open.spotify.com\/playlist\/") != -1) {
+    var start = url.lastIndexOf('/') + 1;
+    var end = url.length;
+    if (url.search('\\?') != -1) {
+     console.log('? detected');
+     end = url.indexOf('?', start);  
+    }
+    return url.slice(start, end);
+  } else {
+    console.log(url + "is not a valid Spotify URL");
+    let message = document.getElementById("error-message");
+    message.innerHTML = "Must be a playlist link from Spotify.";
```

Comment:
After this message has appeared, will it go away when you enter a valid URL, or does it stick around once until you refresh the page?

If this bug does exist, you can try following TDD and write a test that catches this bug, and then fix the bug and see the test pass. If it's too difficult though, don't worry about it, but do still fix the bug :)
https://github.com/googleinterns/step/blob/master/walkthroughs/week-5-tdd/intro/tdd-intro-walkthrough.md#test-driven-development

Reply:
@pruess-google Should I do that in this PR or a separate one?

Reply:
A separate PR is fine